### ISSUE TITLE
Build failure: mkdocs-awesome-pages-plugin

### ISSUE THREAD

**Issue by mcrocker** [NixOS/nixpkgs#329191]
### Steps To Reproduce

1. Create shell.nix:
   ```
   # shell.nix
   let
     pkgs = import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/1d9c2c9b3e71b9ee663d11c5d298727dace8d374.tar.gz") {};  # unstable 2024-07-19T18:15:17-0500
     # pkgs = import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/c19d62ad2265b16e2199c5feb4650fe459ca1c46.tar.gz") {}; # unstable 2024-07-21T14:07:44-0500 latest, but tagged with Build Problem, so used the previous one,
   in pkgs.mkShell {
     packages = [
       (pkgs.python3.withPackages (python-pkgs: [
         python-pkgs.mkdocs # Not needed for the failue, but seems logical to have it here,
         python-pkgs.mkdocs-awesome-pages-plugin
       ]))
     ];
   }
   ```
2. run `nix-shell` command,

### Build log

https://gist.github.com/mcrocker/74f7ce884d33011de17b6993c710a270

### Additional context

* mkdocs-awesome-pages-plugin is only available in unstable, so pinned to a recent build hash to ensure reproducibility, but fails with 
* fails the same way for python3, python311 and python312

### Notify maintainers

@phaer 

### Metadata

```console
user@developer ~/g/d-mkdocs (JiraID-bug-description)> nix-shell -p nix-info --run "nix-info -m"                     10:35:05
 - system: `"x86_64-linux"`
 - host os: `Linux 3.10.0-1160.118.1.el7.x86_64, CentOS Linux, 7 (Core), nobuild`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.22.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

```

**Comment by phaer** [comment]
Hello & thanks for the report. Can confirm the package is broken, presumably since the latest python-updates merge but not entirely sure yet. 

Tests were passing fine when the package was added about a month ago. https://github.com/NixOS/nixpkgs/pull/320709

**Comment by phaer** [comment]
Seems to be related to the mkdocs update to 1.6.0. I filed an upstream issue to track https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin/issues/103

**Comment by mcrocker** [comment]
@paher: Thanks very much for looking into this so quickly and passing it on to upstream.

I figured it worked when you originally created it, but that confirmation suggests a work-around until upstream fixes things. So now I just have to figure out how to get a channel hash from that far back. [Nix Channel Status](https://status.nixos.org/) only tracks the most recent few. 🤔

**Comment by mcrocker** [comment]
Unfortunately, building from the merge commit does not solve the problem:

```
  pkgs = import (builtins.fetchGit {
        name = "nixos-unstable-2024-07-09T07:07-05:00"; # PR #320709
        url = https://github.com/nixos/nixpkgs/;
        rev = "9984f09bb1b658a282dda02ff160cf8ae90c8c9d";
  }) {};
```

This shouldn't be too surprising since it still pulls the same version of the plugin, but I figured it would be worth a try as a possible work-around.

**Comment by phaer** [comment]
It's not about the plugins versions, but about the mkdocs version used. If you want to build the plugin with an older mkdocs while using a current nixpkgs, you could do something like the following:

```nix
{ pkgs ? import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/nixos-unstable.tar.gz") {} }: let
  version = "1.5.3";

  python = pkgs.python3.override {
    packageOverrides = self: super: {
      mkdocs = super.mkdocs.overridePythonAttrs (attrs: {
        inherit version;
        src = pkgs.fetchFromGitHub {
          owner = "mkdocs";
          repo = "mkdocs";
          rev = "refs/tags/${version}";
          hash = "sha256-axH4AeL+osxoUIVJbW6YjiTfUr6TAXMB4raZ3oO0fyw=";
        };
      });
      mkdocs-awesome-pages-plugin = super.mkdocs-awesome-pages-plugin.override {
        mkdocs = self.mkdocs;
      };
    };
  };
in {
  inherit (python.pkgs) mkdocs mkdocs-awesome-pages-plugin;
}
```

**Comment by mcrocker** [comment]
I can confirm that this works for the use-case that this bug was originally created to resolve. ✔️ ⭐ 

```nix
let
  pkgs = import (builtins.fetchGit {
       name = "nixos-unstable-2024-07-25T04-27-0500"; # Fixes Bug #329191 for mkdocs-awesome-pages-plugin
       url = https://github.com/nixos/nixpkgs/;
       rev = "11054378584b9ff970b521c05cb432705c1eb0ec";
 }) {};
in pkgs.mkShell {
  packages = [
    (pkgs.python3.withPackages (python-pkgs: [
      python-pkgs.mkdocs
      python-pkgs.mkdocs-awesome-pages-plugin
      # several other packages specific to my use-case that are irrelevant in this context 😉 
    ]))
  ];
}
```

Since this was first tested, there has been another nixpkgs-unstable build reported on [Nix Channel Status](https://status.nixos.org/) that could be used (with the more compact fetchTarball buildtin) instead of the merge commit hash, but the most recent one has a "Build problem" badge, so I stuck with the merge hash for reporting.

@phaer: Thanks very much for getting this done so quickly.  Also the work-around derivation snippet you provided was above and beyond the call of duty. I would not have been able to get that syntax right myself at this stage in my Nix journey. Unfortunately, it conflicted with some other plugins I was using, but I still learned a lot from it. Thanks again, I appreciate your efforts 🥇 .

**Comment by phaer** [comment]
@mcrocker Glad it helped, it can be rough to know where to start! 

Since the update has been merged to master only yesterday, but it's already in nixpkgs-unstable! When hydra has finished to run nixos tests, nixos-unstable will be updated to include the fix for mkdocs-awesome-plugin automatically. One can check which branches it hit already at https://nixpk.gs/pr-tracker.html?pr=330113
